torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -657,23 +657,20 @@ def get_attr(
         ],
         fx_graph_module: torch.fx.GraphModule,
     ):
-        current_attr = fx_graph_module
-        sub_attr_names = node.target.split(".")  # type: ignore[union-attr]
-        # If node.targe is "conv.weight", the following loop first
-        # assigns fx_module_with_metadata.conv to current_attr, and then
-        # fx_module_with_metadata.conv.weight to current_attr.
-        while sub_attr_names:
-            sub_attr_name = sub_attr_names.pop(0)
-            if not hasattr(current_attr, sub_attr_name):
-                raise AttributeError(
-                    f"Attribute {sub_attr_name} is not found in {current_attr}."
-                )
-            current_attr = getattr(current_attr, sub_attr_name)
-
-        input_ = onnxscript_graph.add_initializer(node.name, current_attr)
+        if not isinstance(node.target, str):
+            raise TypeError(
+                f"node.target must be a str, not {type(node.target)}. Found for node {node}."
+            )
+        assert isinstance(node.target, str), f"node.target {node.target} is not a str."
```

Comment:
the check on previous line `if not isinstance(node.target, str):` already handles that case

Reply:
Huh I meant to delete the previous block but forgot to do it. Since we are here, which version of the two do you prefer?

Reply:
no strong preference